dex/fiatrates/types.go

```diff
@@ -0,0 +1,82 @@
+package fiatrates
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/dex"
+)
+
+// reactivateDuration is how long it'll take before an auto disabled rate source
+// is auto re-enabled.
+var reactivateDuration = 24*time.Hour + FiatRateDataExpiry
+
+type Config struct {
+	CryptoCompareAPIKey string `long:"ccdataapikey" description:"This is your free API Key from cryptocompare.com."`
+	EnableBinanceUS     bool   `long:"enablebinanceus" description:"Set to true, if running the tatanka mesh from a US based server."`
+	DisabledFiatSources string `long:"disabledfiatsources" description:"A list of disabled sources separated by comma. See fiatrate/sources.go."`
+	Assets              string `long:"assets" description:"A list of comma separated assets to fetch rates for when the oracle is activated."`
+}
+
+type source struct {
+	name     string
+	mtx      sync.RWMutex
+	rates    map[string]float64
+	disabled bool
+	// canReactivate is set to false if source was disabled by admin.
+	canReactivate   bool
+	lastRefresh     time.Time
+	requestInterval time.Duration
+	getRates        func(ctx context.Context, assets []string, log dex.Logger) (map[string]float64, error)
+}
+
+func (s *source) isDisabled() bool {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return s.disabled
+}
+
+func (s *source) hasAssets() bool {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return len(s.rates) != 0
+}
+
+func (s *source) isExpired() bool {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return time.Since(s.lastRefresh) > FiatRateDataExpiry
+}
+
+func (s *source) deactivate() {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	s.rates = nil // empty previous rates
+	s.disabled = true
+}
```

Comment:
Definitely gonna want to comment on these various indicators and why they would be used. Or comment in the `source` definition.

Reply:
Missed this in: https://github.com/decred/dcrdex/pull/2736/commits/8f9328ff68c76e3f5de337db2d8218be285078eb 

Will circle back to this.

Reply:
Updated in https://github.com/decred/dcrdex/pull/2736/commits/dbe0d1bf43a49d89df60fa84af09b668f505355f